src/components/StorageOverview/TopConsumers/TopConsumers.js

```diff
@@ -90,30 +98,129 @@ const TopConsumersBody = ({ topConsumerStats }) => {
-  <DashboardCard>
-    <DashboardCardHeader>
-      <DashboardCardTitle>Top Projects by Requested Capacity</DashboardCardTitle>
-    </DashboardCardHeader>
-    <DashboardCardBody isLoading={!topConsumerLoaded} LoadingComponent={LoadingComponent}>
-      <TopConsumersBody topConsumerStats={topConsumerStats} />
-    </DashboardCardBody>
-  </DashboardCard>
-);
+export class TopConsumers extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: metricTypes[0],
+      sortBy: sortBy[0],
+    };
+  }
+
+  changeState = (key, newVal) => {
+    this.setState({ [key]: newVal }, this.getCurrentMetric);
+  };
+
+  getCapacity = data => get(data, 'data.result', []);
+
+  getCurrentMetric = () => {
+    switch (this.state.type) {
+      case PROJECTS:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return this.getCapacity(this.props.projectsUsedCapacity);
+          case BY_REQUESTED_CAPACITY:
+            return this.getCapacity(this.props.projectsRequestedCapacity);
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return [];
+        }
+
+      case STORAGE_CLASSES:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return [];
+          case BY_REQUESTED_CAPACITY:
+            return this.getCapacity(this.props.slClassesRequestedCapacity);
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return [];
+        }
+
+      case PODS:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return this.getCapacity(this.props.podsUsedCapacity);
+          case BY_REQUESTED_CAPACITY:
+            return [];
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return [];
+        }
+      default:
+        // eslint-disable-next-line no-console
+        console.log(`Unknown metric type ${this.state.type.name}`);
+        return [];
+    }
+  };
+
+  render() {
+    const {
+      LoadingComponent,
+      projectsUsedCapacity,
+      projectsRequestedCapacity,
+      slClassesRequestedCapacity,
+      podsUsedCapacity,
+    } = this.props;
+    const metric = this.getCurrentMetric();
+    const isLoading =
+      !projectsUsedCapacity && !podsUsedCapacity && !projectsRequestedCapacity && !slClassesRequestedCapacity;
+
+    return (
+      <DashboardCard>
+        <DashboardCardHeader>
+          <Row>
+            <Col lg={4} md={4} sm={4} xs={4}>
+              <DashboardCardTitle>Top Consumers</DashboardCardTitle>
+            </Col>
+            <Col lg={8} md={8} sm={8} xs={8}>
+              <DashboardCardDropdownBody>
+                <DashboardCardDropdownItem
```

Comment:
IMO this should be disabled if the data isLoading

Reply:
ack